File: backend/src/main/java/com/example/evaluacion2/libros/entity/LibroEntity.java
package com.example.evaluacion2.libros.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "libros")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class LibroEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    
    @Column(nullable = false)
    private String titulo;
    
    @Column(nullable = false)
    private String autor;
    
    @Column(nullable = false)
    private Integer paginas;
    
    @Column(columnDefinition = "TEXT")
    private String descripcion;
    
    private String imagenUri;
    
    @Column(nullable = false)
    private Double precio;
}

File: app/src/main/java/com/example/evaluacion2/data/entities/LibroEntity.kt
package com.example.evaluacion2.data.entities

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "libros")
data class LibroEntity(
    @PrimaryKey(autoGenerate = true) val id: Int = 0,
    val titulo: String,
    val autor: String,
    val paginas: Int,
    val descripcion: String,
    val imagenUri: String = "",
    val precio: Double = 0.0
)
